src/rpp/rpp/utils/utilities.hpp

```diff
@@ -85,4 +85,25 @@ class copy_assignable_callable
 private:
     RPP_NO_UNIQUE_ADDRESS std::optional<Callable> m_callable;
 };
+
+/**
+ * \brief Calls passed function during destruction
+ */
+template<std::invocable Fn>
+class finally_action
+{
+public:
+    finally_action(Fn&& fn)
+        : m_fn{std::move(fn)} {}
```

Comment:
For my education, why don't you need a `std::forward()` inside the `std::move()`? Is it bc move just cast whatever given value to rvalue reference?

Reply:
Okay, let's start again =)))
```cpp
struct A
{
// it just function that accepts just int. It would be copied or moved when function called
void func(int){}
};

struct A
{
// it is rvalue-reverence. int MUST be moved when function called
void func(int&&){}
};

// It is template class. So, when you construct A you MUST to specify it with some SPECIFIC type. For example, if A<int>, then func accepts int via copy/move. If A<int&&>, then rvalue  reference-> must be moved in.
template<typename T>
struct A
{
   void func(T) {}
}

// same as before, but func REQUIRES to accept T by rvalue reference in any case. So, A<int> and A<int&&> are same for us
template<typename T>
struct A
{
   void func(T&&) {}
}

struct A
{
// it just function that accepts just any type. It would be copied or moved when function called. Except of specializing type manually
template<typename T>
void func(T){}
}

struct A
{
// it just function that accepts just any type by reference
template<typename T>
void func(const T&){}
}

struct A
{
// it just function that accepts just any type by forwarding/universal/perfect reference. So, final type of T would be deduced based on passed object. if you pass func(int{}), then T is just int. if you pass func(my_variable), then T is just int& and references collapsing rule collapse it to just reference.
template<typename T>
void func(T&&){}
}

template<typename T>
struct A
{
// same as before
template<typename U>
void func(U&&){}
}
```

Reply:
As a result, in this case it just accepts RVALUE to Fn, not forwarding reference